docs/standard/serialization/system-text-json-how-to.md

```diff
@@ -698,6 +802,118 @@ With this option, the `Summary` property of the `WeatherForecastWithDefault` obj
 
 Null values in the JSON are ignored only if they are valid. Null values for non-nullable value types cause exceptions.
 
+## Preserve references and handle circular references
+
+::: zone pivot="dotnet-5-0"
+
+To preserve references and handle circular references, set <xref:System.Text.Json.JsonSerializerOptions.ReferenceHandler%2A> to <xref:System.Text.Json.Serialization.ReferenceHandler.Preserve%2A>. This setting causes the following behavior:
+
+* On serialize:
+
+  When writing complex types, the serializer also writes metadata properties (`$id`, `$values`, and `$ref`).
+
+* On deserialize:
+
+  Metadata is expected (although not mandatory), and the deserializer tries to understand it.
+
+The following code illustrates use of the `Preserve` setting.
+
+:::code language="csharp" source="snippets/system-text-json-how-to-5-0/csharp/PreserveReferences.cs" highlight="34":::
+
+Here are some other notes about preserving references:
+
+* `MaxDepth` validation is not affected.
+* The following types can be serialized as references but can't be deserialized as references:
+  * Immutable types
+  * Arrays
+  * Value types, such as structs
+* To determine if objects are equal, `System.Text.Json` uses the default equality of `Hashset` or `Dictionary`, which calls <xref:System.Object.GetHashCode%2A?displayProperty=nameWithType> or <xref:System.Object.Equals%2A?displayProperty=nameWithType>.
+* An exception is thrown if there are properties other than `$ref` in a JSON reference object.
```

Comment:
> What does "JSON reference object" mean?

We should probably say "JSON object that contains a $ref metadata" since that's what we use on the [`ReferenceHandler.Preserve`](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.serialization.referencehandler.preserve?view=net-5.0) notes.

Those other three bullets that you mentioned are already specified in said doc.

Reply:
Nice, we should link to this doc then: https://docs.microsoft.com/en-us/dotnet/api/system.text.json.serialization.referencehandler.preserve?view=net-5.0.

Reply:
We should at least link to [ReferenceHandler.Preserve](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.serialization.referencehandler.preserve?view=net-5.0) in this doc.